packages/native/types/base.d.ts

```diff
@@ -82,7 +82,7 @@ export interface StyledComponent<
   ): StyledComponent<
     ComponentProps & React.ComponentProps<C>,
     {},
-    { ref?: React.Ref<InstanceType<C>> }
+    { ref?: React.Ref<InstanceType<C> | undefined> }
```

Comment:
hm, this should pretty much be the same thing - there is little to no difference between an optional property and a property that accepts undefined (and I wish that there was a difference 😢 )

could you add a test case that would show what exactly has been fixed?

Reply:
Ok, I see you have put one in the description of the PR. I think this works exactly the same as `@types/react`. Take a look at this:
https://codesandbox.io/s/keen-frog-39lxt?file=/src/App.tsx:170-173
and the version that fixes the problem:
https://codesandbox.io/s/ecstatic-meninsky-3wzlh?file=/src/App.tsx:174-177

The key is to provide the initial value of `null` to `useRef`. It reflects React expectation (refs are set to either the host element or null)

Reply:
You're right.